### PR TITLE
Assign empty filelist

### PR DESCRIPTION
**Pull request by hekkim**
First of all, I'm really sorry about that I made a mistake about handling `<input type="file" />` in the previous PR (#835).

So, when you try to clean up files through `setValue`, you can see the following error because input's files only accept `FileList`.
```
Uncaught TypeError: Failed to set the 'files' property on 'HTMLInputElement': The provided value is not of type 'FileList'.
```

I just changed `input.files` assigned empty `FileList` when user pass `null` or `undefined`.

I'm sorry again for the previous mistake.

ps. I would like to add test case for sure, but it looks like there's no way to create `FileList` manually in node environment. If I find the way, I'll add test cases.
ps2. If you think user needs to pass empty `FileList` manually to `setValue` for removing files, I'm going to change the codes to accept only `FileList`.

### CODE REVIEW

**Comment by bluebill1049** [comment]
that's all good, perhaps follow with a quick codesandbox with your solution above and i can do a quick test as well. it's my bad as well which i didn't test out your solution last time. all good tho :) we can patch it soon.

**Comment by bluebill1049** [comment]
Looking at stack overflow looks like set empty string is correct https://stackoverflow.com/questions/20549241/how-to-reset-input-type-file

**Comment by hekkim** [comment]
Here's a codesandbox example.

https://codesandbox.io/s/sleepy-currying-duzgg

Reproduce scenario.

1. Attach an image.
2. Click Delete button.
3. Show an error.

The difference with codes in stackoverflow that you found is we're assign FileList into `files` instead of `value`.
So, the error is occurred because we're trying to assign invalid type of value.

**Comment by bluebill1049** [comment]
awesome i will do a quick test with your code tonight and verify the fix 👍

File: src/useForm.ts [hunk 1]
```diff
@@ -194,6 +194,8 @@ export function useForm<FormValues extends FieldValues = FieldValues>({
       } else if (isFileInput(type)) {
         if (value instanceof FileList) {
           ref.files = value;
+        } else {
+          ref.files = new DataTransfer().files;
```
Comment (bluebill1049) [inline src/useForm.ts]:
```suggestion
          ref.value = value;
```

Comment (bluebill1049) [reply src/useForm.ts]: let's do that, it's probably better if user just want to set the value of the input.

Comment (hekkimmt) [reply src/useForm.ts]: Cool. It works very well.
I'm gonna push changes very soon.

Comment (bluebill1049) [reply src/useForm.ts]: you can click above with commit suggestion :)

**Review by bluebill1049 — APPROVED** [review]

File: src/useForm.ts (hunk 1 above)
Comment (hekkimmt) [reply src/useForm.ts]: Oops. I just updated it again.
It's my first time to contribute codes in open source, so I'm a little awkward to handle it. 😅 
From the next time, I'll follow the direction like you said.

**Pull request merged**